streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -185,27 +209,18 @@ public void shouldSendWithNoPartition() {
         assertEquals(3L, (long) offsets.get(new TopicPartition(topic, 0)));
         assertEquals(2L, (long) offsets.get(new TopicPartition(topic, 1)));
         assertEquals(1L, (long) offsets.get(new TopicPartition(topic, 2)));
-        assertEquals(9, producer.history().size());
+        assertEquals(9, mockProducer.history().size());
     }
 
     @Test
     public void shouldUpdateOffsetsUponCompletion() {
-        final RecordCollector collector = new RecordCollectorImpl(
```

Comment:
We can avoid to create "collectors" in almost all tests now reducing the overall complexity of the test